crates/recent_projects/src/remote_servers.rs

```diff
@@ -301,6 +301,43 @@ impl Mode {
         Self::Default(DefaultState::new(cx))
     }
 }
+/// Initialize app-level registration for OpenRemote
+pub fn init(cx: &mut App) {
+    // Register app-level action for OpenRemote
+    cx.on_action({
+        move |_: &OpenRemote, cx: &mut App| {
+            if let Some(app_state) = AppState::try_global(cx).and_then(|weak| weak.upgrade()) {
+                // If window exists, use it
+                if let Some(window) = cx.active_window() {
+                    if let Some(workspace_window) = window.downcast::<Workspace>() {
```

Comment:
Let's add a static `fn .. -> Option<Workspace>` for all this extractions, as we might want to invoke things headless for ~10 different actions it seems.